Java/parkservices/src/main/java/com/amazon/randomcutforest/parkservices/state/predictorcorrector/PredictorCorrectorState.java

```diff
@@ -37,6 +38,10 @@ public class PredictorCorrectorState implements Serializable {
     private long randomSeed;
     private double noiseFactor;
     private boolean autoAdjust;
+    private boolean ignoreDrift;
+    private ComputeDescriptorState lastDescriptor;
+    private int runLength;
+    private double samplingSuppport;
```

Comment:
ParkServices did not exist in https://github.com/aws/random-cut-forest-by-aws/tree/2.0/Java.
For ParkServices we have some tests in https://github.com/aws/random-cut-forest-by-aws/blob/main/Java/parkservices/src/test/java/com/amazon/randomcutforest/parkservices/state/V2TRCFToV3StateConverterTest.java

for ThresholdedRandomCutForest. I can see (with a breakpoint)  the initial few lines ...

ThresholdedRandomCutForestState(version=2.1, forestState=RandomCutForestState(version=2.0, totalUpdates=1257, timeDecay=1.0E-4, numberOfTrees=30, sampleSize=256, shingleSize=8, dimensions=32, outputAfter=32, compressed=true, partialTreeState=true, boundingBoxCacheFraction=0.0, storeSequenceIndexesEnabled=false, compact=true, internalShinglingEnabled=false, centerOfMassEnabled=false, precision=FLOAT_32, pointStoreState=PointStoreState(version=2.0, dimensions=32, capacity=7681, shingleSize=8, precision=FLOAT_32, 

So that looks like a pretty old model. RCF 3.0 (and of course -rc3) started with this numbering of TRCF. So we should be ok with all the changes in ParkServices. 

Now, we do have https://github.com/aws/random-cut-forest-by-aws/blob/3.0-rc3-java/Java/serialization/src/main/java/com/amazon/randomcutforest/serialize/json/v2/V2StateToV3ForestConverter.java
in the serialization package and that has not been changed. So it if it worked earlier (this converter was invoked) then it should be ok -- but I see no evidence/test that it worked earlier. Let me dig into it more.

Reply:
AFAI can tell the models in the V2StateToV3ForestConverterTest are 3.0-rc.1, because the version of RandomCutForestState was upgraded in 3.0-rc2. The model in https://github.com/opensearch-project/anomaly-detection/blob/main/src/test/resources/org/opensearch/ad/ml/rc1_trcf_model_direct.json correspond to version bump to 3.0-rc2 and also pass the test when added to the byte_64 resources. Let me know if I should add it (it's a chunk, so do not want to add to the size unless needed). Many other checkpoints in OpenSearch AD have no versioning and look like version 1.0 of RCF. It may be wiser to rebuild models for such old versions.